### PR TITLE
Add support for import type

### PR DESCRIPTION
**Pull request by arv**
And make sure the import/export validation works.

import type is a flow extension.

### CODE REVIEW

File: test/feature/Modules/Error_ImportType.module.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Options: --types
+// Error: :4:14: 'missing' is not exported by 'test/feature/Modules/resources/default-and-named-type.js'
```
Comment (johnjbarton) [inline test/feature/Modules/Error_ImportType.module.js]: Can we say `type 'missing' is not exported` here?

File: test/feature/Modules/ImportType.module.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// Options: --types
+
+import type {y} from './resources/named-type.js';
+import type def, {x} from './resources/default-and-named-type.js';
```
Comment (johnjbarton) [inline test/feature/Modules/ImportType.module.js]: Do we need to verify  `def` is `string`?

**Comment by johnjbarton** [comment]
LGTM

File: test/feature/Modules/Error_ImportType.module.js (hunk 1 above)
Comment (arv) [inline test/feature/Modules/Error_ImportType.module.js]: I looked at it and it is not obvious. The way flow treats exports as types requires program flow analysis.

```js
class C {}
var x = C;
export {x as y};
```

Here the module exports y as a function but also as a type.

File: test/feature/Modules/ImportType.module.js (hunk 2 above)
Comment (arv) [inline test/feature/Modules/ImportType.module.js]: Not sure what you mean here?

**Pull request merged**